src/check_release.rs

```diff
@@ -157,39 +158,35 @@ pub(super) fn run_check_release(
     let mut results_with_errors = vec![];
     for (semver_query, time_to_decide, results) in all_results {
         config
-            .log_verbose(|config| {
+            .log_verbose(|_| {
```

Comment:
I don't think the callbacks will need config anymore, so we could (1) make the methods `log_*` take only `&self` if we go with this architecture, (2) move the `shell_print_*` methods to (associated) functions, and/or (3) remove the `config` parameter from the closure argument

Reply:
It might be a good idea to let the `config` variable carry explicit `Box<dyn std::io::Write>` fields for `stdout` and `stderr`, and pass them here.

Imagine a user is using `cargo-semver-checks` as a library. It's a bit strange if a library is printing to *it's parent program's* stdout/stderr, right? It would be nice if the parent program could configure where and how it wants the output printed, and then decide by itself whether to actually put it on stdout/stderr.

Reply:
> Imagine a user is using `cargo-semver-checks` as a library. It's a bit strange if a library is printing to _it's parent program's_ stdout/stderr, right? It would be nice if the parent program could configure where and how it wants the output printed, and then decide by itself whether to actually put it on stdout/stderr.

On `main` branch, I don't think there's a mechanism to set the output streams programmatically.  Is that something we want? I can add it in this PR, or a different one.

Reply:
> they won't configure a separate stdout/stderr for logging

Not configure separate stdout/stderr, just give them the option to redirect the cargo-semver-checks-as-library output to a buffer (or drop it altogether) instead of having it printed to *their app's* stdout/stderr.

> I can add it in this PR, or a different one.

Not worth adding here, let's optimize this PR for being easy to review and merge. We can add that in the future, I just wanted to not remove that option from us. If this PR can make sure to print to the configured `Box<dyn Write>` values (or whatever other representation is best), we can handle the rest later.